src/io/flutter/sdk/FlutterSdkManager.java

```diff
@@ -5,71 +5,73 @@
  */
 package io.flutter.sdk;
 
-import com.google.common.base.Objects;
 import com.intellij.openapi.components.ServiceManager;
 import com.intellij.openapi.project.Project;
 import com.intellij.openapi.project.ProjectManager;
 import com.intellij.openapi.project.ProjectManagerAdapter;
-import com.intellij.openapi.roots.RootProvider;
-import com.intellij.openapi.roots.impl.libraries.ApplicationLibraryTable;
+import com.intellij.openapi.roots.impl.libraries.ProjectLibraryTable;
 import com.intellij.openapi.roots.libraries.Library;
 import com.intellij.openapi.roots.libraries.LibraryTable;
+import com.intellij.openapi.util.Disposer;
 import com.intellij.util.EventDispatcher;
 import org.jetbrains.annotations.NotNull;
 
 import java.util.EventListener;
+import java.util.Timer;
+import java.util.TimerTask;
 
 /**
  * Monitors the application library table to notify clients when Flutter SDK configuration changes.
  */
 public class FlutterSdkManager {
   private final EventDispatcher<Listener> myDispatcher = EventDispatcher.create(Listener.class);
-  private final LibraryTableListener myLibraryTableListener = new LibraryTableListener();
-  private final RootProvider.RootSetChangedListener rootListener = x -> checkForFlutterSdkChange();
   private boolean isFlutterConfigured;
-  private final Project myProject;
 
   @NotNull
   public static FlutterSdkManager getInstance(@NotNull Project project) {
     return ServiceManager.getService(project, FlutterSdkManager.class);
   }
 
-  private FlutterSdkManager(Project project) {
-    myProject = project;
+  private FlutterSdkManager(@NotNull Project project) {
+    final LibraryTableListener libraryTableListener = new LibraryTableListener(project);
+    ProjectLibraryTable.getInstance(project).addListener(libraryTableListener);
 
-    listenForSdkChanges();
-    // Cache initial state.
-    isFlutterConfigured = isFlutterSdkSetAndNeeded();
-  }
+    final Timer timer = new Timer();
+    timer.scheduleAtFixedRate(new TimerTask() {
```

Comment:
Polling like this is a battery drain for laptop users, so I would expect at least a TODO here to switch to using real listeners again.